client/app/components/packages/pas-form/edit.js

```diff
@@ -30,7 +30,14 @@ export default class PasFormComponent extends Component {
 
   @action
   async savePackage() {
-    await this.args.package.save();
+    try {
+      await this.args.package.save();
+      await this.args.package.reload();
+    } catch (error) {
+      console.log('Save package error:', error);
+    }
+
+    this.args.package.refreshExistingDocuments();
```

Comment:
I think this works for now, but I'm beginning to think that these documents should be their own model now. This line here is a sign that we're sort of having to manually manage the state of documents. Right? For example, when we have docs queued up for upload, those are "dirty" in a way, we can show those as such. When they save, their "clean", they've saved, we can show those accordingly.

I think a refactor task should be to seriously re-think doc upload and embrace Ember Data for them.

Reply:
That makes a lot of sense -- I'm going to go ahead and merge this for the sake of the end of the sprint and then I'll be sure to make an issue addressing this, thanks!